### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by minhtranite** [ckeditor/ckeditor5#1341]
## Is this a bug report or feature request? (choose one)

🐞 Bug report 

## 💻 Version of CKEditor

<script src="https://cdn.ckeditor.com/ckeditor5/11.1.1/balloon/ckeditor.js"></script>

## 📋 Steps to reproduce

1. Create a editor instance
2. Destroy it

## ✅ Expected result

Memory not increment

## ❎ Actual result

Memory increment each time create and destroy

## 📃 Other details that might be useful

- Codepend: https://codepen.io/minhtranite/pen/Xybegm?editors=1010
- Screenshot video: https://drive.google.com/file/d/17Hqae0B60V8BR0Cq0CjiVJL7-cgc6R7S/view?usp=sharing

**Comment by oleq** [comment]
I also confirm that:

![image](https://user-images.githubusercontent.com/1099479/48063245-b7e6dc80-e1c4-11e8-9714-fa98fa0d5ce5.png)

![image](https://user-images.githubusercontent.com/1099479/48063256-c1704480-e1c4-11e8-8396-438531da546e.png)

![image](https://user-images.githubusercontent.com/1099479/48063262-c7febc00-e1c4-11e8-9758-a8053d9868e3.png)

After the first destruction

![image](https://user-images.githubusercontent.com/1099479/48063564-bb2e9800-e1c5-11e8-8628-ce4b4fcb5a17.png)

After the second destruction

![image](https://user-images.githubusercontent.com/1099479/48063572-c255a600-e1c5-11e8-9576-003632a0fd1e.png)


cc @Reinmar

**Comment by Reinmar** [comment]
This is a regression because we tested for that about 2 years ago and there were no leaks at all.

**Comment by oleq** [comment]
> about 2 years ago

I think that's why we should make it a part of every major release.

**Comment by Reinmar** [comment]
Yes. We should have a regression test for that. As a general rule – since we have a bug, we must have a test. Ideally – automated. AFAIR there was some API available for making snapshots and accessing such performance information. Even if it's in just one browser, let's have a test for that.

**Comment by minhtranite** [comment]
Sorry! What is `iteration 22` milestone mean ? It not assign due date so I want to ask when this issue can be fixed. I'm using it on production app so It will be good if you can fix it soon.

**Comment by jodator** [comment]
@minhtranite Hi the good news is that we identified what was the problem and it will be fixed soon.

The iteration 22 is a current one so the next official release should have this fix included.

**Comment by jodator** [comment]
@Reinmar & @pomek: Some technical question about Travis integration. The tests uses Google Chrome run with additional flag in order to expose garbage collector for tests. The tests itself are generic one that runs create/destroy couple of times and compares used JS heap size after first run and after the last.

After some fiddling with them locally I've identified that 1 MB is sufficient to both minimize false positives and to actually catch a memory leaks automatically. Unfortunately memory measurements are repeatable to some extent. To minimize errors the garbage collecting step is necessary.

Also I think that running memory leak tests only for Chrome is sufficient to detect them.

The question is how to enable them on CI? Note: there will be dedicated manual test pages with instructions on how to check memory usage.

We can either:
- expose garbage collector for all the tests run with Chrome ([add a flag to current Chrome configuration](https://github.com/ckeditor/ckeditor5-dev/commit/2fc8dc430ffb2de60113d6eff89fabdbc46ada3e))
- or create dedicated step which will run only memory related tests with dedicated Chrome configuration

**Comment by jodator** [comment]
> After some fiddling with them locally I've identified that 1 MB is sufficient to both minimize false positives

There was one more component causing memory leaks. Probably we're going to be safe with `0` as a test there.

**Comment by minhtranite** [comment]
Thank you guys!
